sql/catalyst/src/test/scala/org/apache/spark/sql/types/decimal/DecimalSuite.scala

```diff
@@ -46,11 +46,8 @@ class DecimalSuite extends SparkFunSuite with PrivateMethodTester {
     checkDecimal(Decimal(1e17.toLong, 18, 0), 1e17.toLong.toString, 18, 0)
     checkDecimal(Decimal(Long.MaxValue), Long.MaxValue.toString, 20, 0)
     checkDecimal(Decimal(Long.MinValue), Long.MinValue.toString, 20, 0)
-    intercept[IllegalArgumentException](Decimal(170L, 2, 1))
```

Comment:
This patch has change to use Decimal(unscaleLong, precision, scale) only for serialization, and the caller will make sure that it will not overflow.